### ISSUE TITLE
[FEATURE] Store the audit log history

### ISSUE THREAD

**Issue by coolderli** [apache/gravitino#2061]
### Describe the feature

The audit log history is significant. It will help us to know who performed what operation at what time. Now we have already the `AuditInfo`, but we only record the latest operation. Should we store all audit history in Gravitino? Or we can directly hand it over to the user for implementation.

If we keep all audit history in the Gravitino Kv store, it will cost a  large amount of storage space. And we have to handle with the expired data. 

I propose this idea in the hope that more people can discuss it.

### Motivation

Support audit log history

**Comment by FANNG1** [comment]
Yes, it's useful to save audit history, one possible solution is to provide a general event listener system for DDL operations. we could provide an auditInfo event consumer to save auditInfo and users could provide the custom event consumer to satisfy the internal environment. But this solution has the risk of losing the event because the event consumer is async.
